### DISCUSSION TITLE
How to parse body for Upsertable Object ?

### DISCUSSION THREAD

**Discussion by VaZark** [prisma/prisma#13639] [Q&A]
I'm currently parsing body directly using `as Product`, however when i do the upsert function, it throws an error `The "string" argument must be of type string or an instance of Buffer or ArrayBuffer. Received an instance of PrismaClientValidationError`.

So it looks something like 

```prisma

model Product {
  id                String              @id @default(cuid())
  programme         String              @unique @db.VarChar(200)
  display_name      String              @db.VarChar(200)
}
```

```node
const  {id, ...body} = req.body as Product
const product = await prisma.product.upsert({
    where: {
        id: id
    },
    update : body ,
    create : body 
})
```

 Is there any type or function that can keep it simple without having to rewriting / cloning the body and skipping the ID ?

**Comment by austincrim** [comment]
Hey @VaZark 👋,

Can you share what the `body` variable actually looks like at runtime?

**Reply by vladi-stevanovic** [reply]
👋 Hello @VaZark  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by VaZark** [reply]
Unfortunately, we decided to not move forward with Prisma after my PoC due to the lack of support for readonly models and SQL Views at that time (not sure if that changed recently). Thus,I have no point of reference regarding how I came across this error. 

So, I am unable to meaningfully contribute anymore.

**Reply by pantharshit00** [reply]
Hello @VaZark,

Thanks for reaching back to us! Your feedback is quite valuable. 

We are currently building the SQL Views functionality. There is an active survey running on this topic where we are seeking for feedback. You can find more details here if you are interested: https://github.com/prisma/prisma/issues/678#issuecomment-1330304851

We will be happy to assist you future if you decide to do any other project using Prisma.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
